src/text_to_binary/rewrite.ml

```diff
@@ -407,6 +407,131 @@ let rewrite_types (_modul : Assigned.t) (t : binary str_type) :
   let t = [ (None, (Final, [], t)) ] in
   Ok t
 
+let rec rewrite_term (binder_list : string option list)
+  ?(modul : Binary.modul = Binary.empty_modul)
+  ?(func_param_list : string option list = []) :
+  text Spec.term -> binary Spec.term Result.t =
+  let rec aux error acc id = function
+    | [] -> Error error
+    | Some id' :: bl ->
+      if String.equal id id' then Ok (Raw acc) else aux error (acc + 1) id bl
+    | _ :: bl -> aux error (acc + 1) id bl
+  in
+
+  let find_binder (binder_list : string option list) (ind : text indice) :
+    binary indice Result.t =
+    match ind with
+    | Raw id -> Ok (Raw id)
+    | Text id -> aux (`Unknown_binder ind) 0 id binder_list
+  in
+
+  let find_param (func_param_list : string option list) (ind : text indice) :
+    binary indice Result.t =
+    match ind with
+    | Raw id -> Ok (Raw id)
+    | Text id -> aux (`Unknown_param ind) 0 id func_param_list
+  in
+
+  let find_global (modul : Binary.modul) (ind : text indice) :
+    binary indice Result.t =
+    match ind with
+    | Raw id -> Ok (Raw id)
+    | Text id -> (
+      match String_map.find_opt id modul.global.named with
+      | None -> Error (`Unknown_global ind)
+      | Some i -> Ok (Raw i) )
+  in
+
+  let open Spec in
+  function
+  | Int32 i32 -> Ok (Int32 i32)
+  | Int64 i64 -> Ok (Int64 i64)
+  | Float32 f32 -> Ok (Float32 f32)
+  | Float64 f64 -> Ok (Float64 f64)
+  | Var ind -> (
+    match
+      ( find_binder binder_list ind
+      , find_param func_param_list ind
+      , find_global modul ind )
```

Comment:
OK we'll have to think about this more. This seems reasonable but it may be a little bit dangerous too...

Reply:
In what sense? efficiency?